shapely/geometry/base.py

```diff
@@ -28,6 +29,16 @@
 ]
 
 
+def geom_factory(g, parent=None):
+    warn(
```

Comment:
`parent` is unused?  Can it be removed, or does it need to be retained for backward compatibility?

Should this also include a brief docstring, something like

```Python
"""Creates a Shapely geometry instance from a pointer to a GEOS geometry.

WARNING: the GEOS library used to create the the GEOS geometry pointer and the GEOS library used by Shapely must be exactly the same, or unexpected results or segfaults may occur.

Deprecated in Shapely 2.0

"""
```

Reply:
Yes, I only added it back for backwards compatibility. Apart from keeping a reference to that parent object, it didn't really do anything (https://github.com/shapely/shapely/blob/b6aa26a0bcd9cb1f0767d5d80df0fe32eacd5b31/shapely/geometry/base.py#L70-L91)

Reply:
> Should this also include a brief docstring, something like

Thanks! That's a good addition